### PR TITLE
Wallet: Fix error after clicking previous page

### PR DESCRIPTION
**Pull request by cehali**
# Summary
When creating an account, clicking the previous page and accepting terms once again generated an error with config variables. Simple fix, the current behavior of the application can be discussed.

Fixes: 1

## Checklist
- [x] Change is small and easy to review (please split big changes into multiple PRs)
- [x] Change is consistent with architecture
- [x] Change is consistent with test architecture
- [x] Change is consistent with naming conventions
- [x] New code is covered with tests
- [x] Tests related to old code are updated
- [x] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-wallet/src/ui/react/Modals/Modal.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ const Modal = () => {
         </ModalWrapperClosable>
       );
     case 'topUpAccount':
-      return (
+      return config ? (
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/Modal.tsx]: Why would config be null?

Comment (cehali) [reply universal-login-wallet/src/ui/react/Modals/Modal.tsx]: When we click the previous button on topUp modal, we see terms. Accepting them moves us to topUp modal once again (skipping choosing name). Modal component has the function 'useRelayerConfig' which uses 'useAsync'. When we click the previous button and then go to topUp modal, modal component tries to get config but 'useRelayerConfig' is not resolved yet and this is why we have an undefined config for short period of time. 'useAsync' rerenders the page and 'useRelayerConfig' is almost instant so the user is not aware of rerendering.

Another option is to wait until 'config' isn't undefined and render some loader but a short period of time, which would be more like blink than loading.

Comment (marekkirejczyk) [reply universal-login-wallet/src/ui/react/Modals/Modal.tsx]: Let's use cache instead

File: universal-login-wallet/src/ui/createServices.ts [hunk 2]
```diff
@@ -44,6 +44,7 @@ export const createServices = (config: Config, overrides : Overrides = {}) => {
     walletService,
     walletPresenter,
     start: () => sdk.start(),
+    getRelayerConfig: () => sdk.getRelayerConfig()
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/createServices.ts]: ech... what is that? is that a service?

File: universal-login-wallet/src/ui/hooks/useRelayerConfig.ts [hunk 3]
```diff
@@ -1,8 +0,0 @@
-import {useAsync} from '@universal-login/react';
-import {useServices} from './useServices';
-
-export const useRelayerConfig = () => {
-  const {sdk} = useServices();
-  const [config] = useAsync(async () => sdk.getRelayerConfig(), []);
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/hooks/useRelayerConfig.ts]: This is actually cool, I would leave it, just update to use `configService`

File: universal-login-wallet/src/core/services/ConfigService.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+export class ConfigService {
+  private sdk: UniversalLoginSDK;
+  public relayerConfig?: PublicRelayerConfig;
+
+  constructor(sdk: UniversalLoginSDK) {
+    this.sdk = sdk;
+  }
+
+  async setRelayerConfig() {
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/core/services/ConfigService.ts]: `fetchRelayerConfig`

File: universal-login-wallet/src/core/services/ConfigService.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    if (!this.relayerConfig) {
+      this.relayerConfig = await this.sdk.getRelayerConfig();
+    }
+  }
+
+  getRelayerConfig() {
+    return this.relayerConfig;
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/core/services/ConfigService.ts]:
```ts
async getRelayerConfig() {
  await fetchRelayerConfig();
  return this.relayerConfig;
}
```

File: universal-login-wallet/src/ui/react/Modals/Modal.tsx [hunk 6]
```diff
@@ -2,18 +2,20 @@ import React, {useContext} from 'react';
...
 import {WalletModalContext} from '../../../core/entities/WalletModalContext';
 import {ConnectionNotificationModal} from '../ConnectAccount/ConnectionNotificationModal';
 
 const Modal = () => {
   const modalService = useContext(WalletModalContext);
-  const {walletPresenter, walletService, sdk} = useServices();
-  const config = useRelayerConfig();
+  const {walletPresenter, walletService, sdk, configService} = useServices();
+  useAsync(async () => configService.setRelayerConfig(), []);
+  const relayerConfig = configService.getRelayerConfig();
```
Comment (marekkirejczyk) [inline universal-login-wallet/src/ui/react/Modals/Modal.tsx]: just use `useRelayerConfig`

**Pull request merged**